### DISCUSSION TITLE
localStorage for selected Language

### DISCUSSION THREAD

**Discussion by d-oit** [gethinode/hinode#1023] [Ideas]
Hello Mark,

first of all great theme. Appreciate your work.

Is there a reason why the selected language is not stored in the localStorage.

Thanks for your reply.

Dominik

**Comment by markdumay** [comment]
Glad you like the theme! Great suggestion to store the selected language - care to submit a PR?

**Reply by d-oit** [reply]
I could give it a try.  Nothing done with Hugo yet. 

I will create a new feature branch for the PR - already have a quick and dirty solution with a critical js file.

**Reply by d-oit** [reply]
PR created. 

The exampleSite not running on my current notebook. I will check it on monday. 

Have a nice weekend!

exampleSite> hugo server
Watching for changes in C:\github\hinode\{archetypes,assets,content,data,exampleSite,i18n,layouts,netlify.toml,package.json,static}
Watching for config changes in C:\github\hinode\exampleSite\config\_default, C:\github\hinode\exampleSite\config\_default\menus, C:\github\hinode\config\_default, C:\github\hinode\config\_default\menus, C:\github\hinode\exampleSite\go.mod, C:\github\hinode\exampleSite\hinode.work
Start building sites … 
hugo v0.127.0-74e0f3bd63c51f3c7a0f07a7c779eec9e922957e+extended windows/amd64 BuildDate=2024-06-05T10:27:59Z VendorInfo=gohugoio

ERROR render of "page" failed: execute of template failed: template: partials/head/head.html:8:7: executing "head" at <partialCached "head/stylesheet-core.html" .>: error calling partialCached: "C:\github\hinode\layouts\partials\head\stylesheet-core.html:3:3": execute of template failed: template: partials/head/stylesheet-core.html:3:3: executing "partials/head/stylesheet-core.html" at <partial "head/stylesheet.html">: error calling partial: "C:\github\hinode\layouts\partials\head\stylesheet.html:72:12": execute of template failed: template: partials/head/stylesheet.html:72:12: executing "partials/head/stylesheet.html" at <partial "head/icons.html" (dict "css" $css)>: error calling partial: "C:\github\hinode\layouts\partials\head\icons.html:22:54": execute of template failed: template: partials/head/icons.html:22:54: executing "partials/head/icons.html" at <$css.Content>: error calling Content: TOCSS-DART: failed to transform "css/main.css" (text/x-scss). You need dart-sass-embedded in your system $PATH.: this feature is not available in your current Hugo version, see https://goo.gl/YMrWcn for more information
ERROR render of "home" failed: execute of template failed: template: partials/head/head.html:8:7: executing "head" at <partialCached "head/stylesheet-core.html" .>: error calling partialCached: "C:\github\hinode\layouts\partials\head\stylesheet-core.html:3:3": execute of template failed: template: partials/head/stylesheet-core.html:3:3: executing "partials/head/stylesheet-core.html" at <partial "head/stylesheet.html">: error calling partial: "C:\github\hinode\layouts\partials\head\stylesheet.html:72:12": execute of template failed: template: partials/head/stylesheet.html:72:12: executing "partials/head/stylesheet.html" at <partial "head/icons.html" (dict "css" $css)>: error calling partial: "C:\github\hinode\layouts\partials\head\icons.html:22:54": execute of template failed: template: partials/head/icons.html:22:54: executing "partials/head/icons.html" at <$css.Content>: error calling Content: TOCSS-DART: failed to transform "css/main.css" (text/x-scss). You need dart-sass-embedded in your system $PATH.: this feature is not available in your current Hugo version, see https://goo.gl/YMrWcn for more information
ERROR TOCSS-DART: failed to transform "css/main.css" (text/x-scss). You need dart-sass-embedded in your system $PATH.: this feature is not available in your current Hugo version, see https://goo.gl/YMrWcn for more information
Built in 1326 ms
Error: error building site: render: failed to render pages: render of "page" failed: execute of template failed: template: partials/head/head.html:8:7: executing "head" at 
<partialCached "head/stylesheet-core.html" .>: error calling partialCached: "C:\github\hinode\layouts\partials\head\stylesheet-core.html:3:3": execute of template failed: template: partials/head/stylesheet-core.html:3:3: executing "partials/head/stylesheet-core.html" at <partial "head/stylesheet.html">: error calling partial: "C:\github\hinode\layouts\partials\head\stylesheet.html:72:12": execute of template failed: template: partials/head/stylesheet.html:72:12: executing "partials/head/stylesheet.html" at <partial "head/icons.html" (dict "css" $css)>: error calling partial: "C:\github\hinode\layouts\partials\head\icons.html:22:54": execute of template failed: template: partials/head/icons.html:22:54: executing "partials/head/icons.html" at <$css.Content>: error calling Content: TOCSS-DART: failed to transform "css/main.css" (text/x-scss). You need dart-sass-embedded in your system $PATH.: this feature is not available in your current Hugo version, see https://goo.gl/YMrWcn for more information

**Reply by markdumay** [reply]
Thanks for taking the time in preparing this PR. Looking at the logs, it appears your local machine is misconfigured:

```bash
You need dart-sass-embedded in your system $PATH.: this feature is not available in your current Hugo
version, see https://goo.gl/YMrWcn for more information
```

Follow these instructions to install Dart Sass locally: https://gohugo.io/hugo-pipes/transpile-sass-to-css/#installing-in-a-development-environment

Alternatively, you can set the transpiler to `libsass` in `params.toml`:

```toml
[main]
    [main.build]
        transpiler = "libsass"
```

**Comment by d-oit** [comment]
BTW:
How to you managed the mixed go var in js with VS Code?  e.g.
{{- if site.Params.main.enableLanguageSelectionStorage -}}

haven't used that before

**Reply by markdumay** [reply]
The script partial uses [ExecuteAsTemplate](https://gohugo.io/hugo-pipes/resource-from-template/) to parse Hugo variables. See the following [code snippet](https://github.com/gethinode/hinode/blob/88dfc50c50c5e12ed3c502f5661886f5c238c69f/layouts/partials/footer/scripts.html#L27C1-L31C12):

```go-template
{{- $bundle := partial "utilities/bundle" (dict "match" $match "filename" $filename "modules" $modules "basepath" "js/modules" "all" true "debugging" site.Params.debugging.showJS) -}}
{{- $js := $bundle -}}
{{- if not $skipTemplate -}}
    {{- $js = $bundle | resources.ExecuteAsTemplate $filename $page -}}
{{- end -}}
```

**Reply by d-oit** [reply]
Interesting - I'll take a look when I get a chance. 

I found your solution with the color.js in the .eslintignore 
Was wondering if the mixing of Hugo template vars in js was solved. But it's not important now that I know how it works 😃
